@xen-orchestra/lite/src/components/AddNetwork.tsx

```diff
@@ -0,0 +1,247 @@
+}
+
+const AddNetwork = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({
+      isBonded: false,
+      isLoading: false,
+      form: {
+        bondMode: '',
+        description: '',
+        mtu: '',
+        nameLabel: '',
+        pifsId: '',
+        vlan: '',
+      },
+    }),
+    computed: {
+      pifs: state => state.objectsByType.get('PIF'),
+      pifsMetrics: state => state.objectsByType.get('PIF_metrics'),
+      collection: state =>
+        state.pifs
+          ?.filter(pif => pif.VLAN === -1 && pif.bond_slave_of === 'OpaqueRef:NULL' && pif.host === pif.$pool.master)
+          .sortBy(pif => pif.device)
+          .valueSeq()
+          .toArray(),
+    },
+    effects: {
+      createNetwork: async function (e) {
+        e.preventDefault()
+        if (this.state.isLoading) {
+          return
+        }
+        this.state.isLoading = true
+        const { bondMode, description, mtu, nameLabel, pifsId, vlan } = this.state.form
+
+        try {
+          await this.state.xapi.createNetworks([
+            {
+              MTU: +mtu,
+              name_description: description,
+              name_label: nameLabel,
+              VLAN: +vlan,
+              bondMode: bondMode === '' ? undefined : bondMode,
+              pifsId: pifsId === '' ? undefined : Array.isArray(pifsId) ? pifsId : [pifsId],
+            },
+          ])
+          this.effects.resetForm()
+        } catch (error) {
+          console.error(error)
+          if (error instanceof Error) {
+            alert({ message: <p>{error.message}</p>, title: <IntlMessage id='networkCreation' /> })
+          }
+        }
+        this.state.isLoading = false
+      },
+      handleChange: function ({ target: { name, value } }) {
+        // Reason why form values are initialized with empty string and not a undefined value
+        // Warning: A component is changing an uncontrolled input to be controlled.
+        // This is likely caused by the value changing from undefined to a defined value,
+        // which should not happen. Decide between using a controlled or uncontrolled input
+        // element for the lifetime of the component.
+        // More info: https://reactjs.org/link/controlled-components
+        const { form } = this.state
+        if (form[name] !== undefined) {
+          this.state.form = {
+            ...form,
+            [name]: value,
+          }
+        }
+      },
+      resetForm: function () {
+        this.state.isBonded = false
+        Object.keys(this.state.form).forEach(property => {
+          this.state.form = {
+            ...this.state.form,
+            [property]: '',
+          }
+        })
+      },
+      toggleBonded: function () {
+        if (Array.isArray(this.state.form.pifsId)) {
+          this.state.form.pifsId = ''
+        } else {
+          this.state.form.pifsId = []
+        }
+        this.state.isBonded = !this.state.isBonded
```

Comment:
if it's an array you set it to a string, and vice versa ? If it's intended, I would like a little comment  or if it's tied to is bonded do something like this : 
```suggestion
        this.state.isBonded = !this.state.isBonded
        // on isBonded change , also reset pifsId
        this.state.form.pifsId = this.state.isBonded ? '' : []
```